arroyo/processing/strategies/dead_letter_queue/policies/produce.py

```diff
@@ -0,0 +1,49 @@
+import json
+
+from arroyo.backends.kafka.consumer import KafkaPayload, KafkaProducer
+from arroyo.processing.strategies.dead_letter_queue.policies.abstract import (
+    DeadLetterQueuePolicy,
+    InvalidMessages,
+)
+from arroyo.types import Topic
+from arroyo.utils.metrics import get_metrics
+
+
+class ProduceInvalidMessagePolicy(DeadLetterQueuePolicy):
+    """
+    Produces given InvalidMessages to a dead letter topic.
+
+    Meant to be used as a baseclass for policies needing to produce
+    invalid messages to a dead letter topic.
+    """
+
+    def __init__(self, producer: KafkaProducer, dead_letter_topic: Topic) -> None:
```

Comment:
Please use an arroyo producer, and not the bare Kafka one
- The arroyo one simplifies the management of callbacks.
- Using the bare Kafka one prevents you from writing unit tests without introducing all the Kafka complexity

Reply:
I'm not sure what you mean by Arroyo producer vs bare Kafka one, `KafkaProducer` is the one we defined in Arroyo here:
https://github.com/getsentry/arroyo/blob/b4ea5a8ecc947610025eaa1c0a35ee363af22777/arroyo/backends/kafka/consumer.py#L659

Do you mean I should use this parent class instead?
https://github.com/getsentry/arroyo/blob/b4ea5a8ecc947610025eaa1c0a35ee363af22777/arroyo/backends/abstract.py#L167

(I'm not using the one from the Kafka library directly if that's what you mean)

Reply:
Sorry I read it wrong, I thought you were using this 
https://github.com/getsentry/arroyo/blob/main/arroyo/backends/kafka/consumer.py#L35

Reply:
Actually this is still not right 
https://github.com/getsentry/arroyo/blob/main/arroyo/backends/abstract.py#L167-L182

Please use the abstract producer instead of the specific kafka one. Otherwise you will not be able to run unit tests of this without relying on Kafka.